<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>The Allele-Specific CNA Analysis of KRAS in CRC â€¢ KrasAlleleCna</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="The Allele-Specific CNA Analysis of KRAS in CRC">
<meta property="og:description" content="This package contains the complete analysis of the copy number
    alteration (CNA) of KRAS in colorectal cancer (CRC).">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">KrasAlleleCna</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/a01-download-getgdcdata.html">A.01 - Downloading Data</a>
    </li>
    <li>
      <a href="articles/b01-dataprep-processannovar.html">B.01 - Parsing ANNOVAR Results</a>
    </li>
    <li>
      <a href="articles/b02-dataprep-processmpileup.html">B.02 - Parsing mpileup Results</a>
    </li>
    <li>
      <a href="articles/c01-calccopynum.html">C.01 - Calculate Allele-specfic Copy Number</a>
    </li>
    <li>
      <a href="articles/d01-plotresults.html">D.01 - Plot Allele-specific Copy Number Results</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    

    
    
<!-- README.md is generated from README.Rmd. Please edit that file -->
<div id="krasallelecna" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#krasallelecna" class="anchor"></a>KrasAlleleCna</h1></div>

<div id="purpose" class="section level2">
<h2 class="hasAnchor">
<a href="#purpose" class="anchor"></a>Purpose</h2>
<p>In <a href="http://cancerdiscovery.aacrjournals.org/content/early/2019/04/05/2159-8290.CD-18-1220">Poulin <em>et al.</em> (2019)</a>, the authors studied the specific oncogenic properties of <em>KRAS</em> A146T mutations, a hotspot-mutation only found in colorectal adenocarcinma (CRC). For this paper, I was asked to find out whether there was a difference in the copy number of the mutant allele between human CRC patients with G12D (the most common <em>KRAS</em> oncogenic mutation) and A146T mutations. Not only would a difference suggest that dosage is an important factor, but could also point to differences in interactions with the wild-type (WT) allele [<a href="https://www.ncbi.nlm.nih.gov/pubmed/27422332">Zhou <em>et al.</em>, 2016</a>, <a href="https://www.ncbi.nlm.nih.gov/pubmed/29425486">Lin and Haigis, 2018</a>].</p>
<p>This analysis was divided into 4 parts: A) downloading the data, B) processing the data, C) calculating the allele copy number, and D) analyzing the results.</p>
</div>
<div id="conclusions" class="section level2">
<h2 class="hasAnchor">
<a href="#conclusions" class="anchor"></a>Conclusions</h2>
<p>There was no detectable difference between the copy number of the G12D or A146T <em>KRAS</em> alleles.</p>
<p>The below plot showed the copy number of the mutant and WT allele in G12D and A146T samples.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">cn_tib &lt;-<span class="st"> </span>allele_data_filt <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/filter">filter</a></span>(project_id <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"TCGA-COAD"</span>, <span class="st">"TCGA-READ"</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="st">    </span><span class="kw">group_by</span>(aa_mod) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">allele_count =</span> <span class="kw">n_distinct</span>(common_id)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">cn_mut =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/ifelse">ifelse</a></span>(cn_mut <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">0</span>, cn_mut),</a>
<a class="sourceLine" id="cb1-7" data-line-number="7">           <span class="dt">cn_wt =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/ifelse">ifelse</a></span>(cn_wt <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">0</span>, cn_wt),</a>
<a class="sourceLine" id="cb1-8" data-line-number="8">           <span class="dt">codon =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw">str_extract</span>(aa_mod, <span class="st">"[:digit:]+"</span>)))</a>
<a class="sourceLine" id="cb1-9" data-line-number="9">cn_tib <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">aa_mod =</span> forcats<span class="op">::</span><span class="kw"><a href="https://forcats.tidyverse.org/reference/fct_reorder.html">fct_reorder</a></span>(aa_mod, codon)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/filter">filter</a></span>(aa_mod <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"A146T"</span>, <span class="st">"G12D"</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">diff =</span> cn_mut <span class="op">-</span><span class="st"> </span>cn_wt) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="st">    </span><span class="kw">ggplot</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="st">    </span><span class="kw">facet_grid</span>(. <span class="op">~</span><span class="st"> </span>aa_mod) <span class="op">+</span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15"><span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="st">"mutant"</span>, <span class="dt">xend =</span> <span class="st">"wild type"</span>,</a>
<a class="sourceLine" id="cb1-16" data-line-number="16">                     <span class="dt">y =</span> cn_mut, <span class="dt">yend =</span> cn_wt,</a>
<a class="sourceLine" id="cb1-17" data-line-number="17">                     <span class="dt">color =</span> diff)) <span class="op">+</span></a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="st">    </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="st">"mutant"</span>, <span class="dt">y =</span> cn_mut, <span class="dt">color =</span> diff), <span class="dt">size =</span> <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="st">    </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="st">"wild type"</span>, <span class="dt">y =</span> cn_wt, <span class="dt">color =</span> diff), <span class="dt">size =</span> <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="st">    </span><span class="kw">scale_color_gradient</span>(<span class="dt">low =</span> <span class="st">"blue"</span>, <span class="dt">high =</span> <span class="st">"red"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb1-21" data-line-number="21"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">panel.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">"white"</span>),</a>
<a class="sourceLine" id="cb1-22" data-line-number="22">          <span class="dt">panel.grid.major =</span> <span class="kw">element_line</span>(<span class="dt">color =</span> <span class="st">"grey85"</span>),</a>
<a class="sourceLine" id="cb1-23" data-line-number="23">          <span class="dt">legend.key.height =</span> <span class="kw">unit</span>(<span class="fl">0.2</span>, <span class="st">"in"</span>),</a>
<a class="sourceLine" id="cb1-24" data-line-number="24">          <span class="dt">legend.key.width =</span> <span class="kw">unit</span>(<span class="fl">0.1</span>, <span class="st">"in"</span>),</a>
<a class="sourceLine" id="cb1-25" data-line-number="25">          <span class="dt">axis.ticks =</span> <span class="kw">element_line</span>(<span class="st">"grey85"</span>),</a>
<a class="sourceLine" id="cb1-26" data-line-number="26">          <span class="dt">strip.background =</span> <span class="kw">element_blank</span>()) <span class="op">+</span></a>
<a class="sourceLine" id="cb1-27" data-line-number="27"><span class="st">    </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">"allele"</span>, <span class="dt">y =</span> <span class="st">"copy number"</span>,</a>
<a class="sourceLine" id="cb1-28" data-line-number="28">         <span class="dt">title =</span> <span class="st">"Copy number of both alleles in KRAS mutants"</span>,</a>
<a class="sourceLine" id="cb1-29" data-line-number="29">         <span class="dt">color =</span> <span class="st">"CN difference"</span>)</a></code></pre></div>
<p><img src="reference/figures/README-lateralplot-1.png" width="100%"></p>
<p>Though there seems to be increased amplification in human tumor samples, the number of available samples limited the power of the study.</p>
<p>Heatmaps of the genotypes of the samples highlighted that the most common genotype was 1 mutant : 1 WT allele.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">plot_genotype_heatmap</span>(cn_tib, <span class="st">"A146T"</span>)</a></code></pre></div>
<p><img src="reference/figures/README-genotypeheatmap-1.png" width="100%"></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">plot_genotype_heatmap</span>(cn_tib, <span class="st">"G12D"</span>)</a></code></pre></div>
<p><img src="reference/figures/README-genotypeheatmap-2.png" width="100%"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Joshua H Cook <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>

      <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://www.gnu.org/licenses/gpl-3.0"><img src="https://img.shields.io/badge/License-GPLv3-blue.svg" alt="License: GPL v3"></a></li>
<li><a href="https://travis-ci.org/jhrcook/KrasAlleleCna"><img src="https://travis-ci.org/jhrcook/KrasAlleleCna.svg?branch=master" alt="Travis build status"></a></li>
<li><a href="https://ci.appveyor.com/project/jhrcook/KrasAlleleCna"><img src="https://ci.appveyor.com/api/projects/status/github/jhrcook/KrasAlleleCna?branch=master&amp;svg=true" alt="AppVeyor build status"></a></li>
<li><a href="https://codecov.io/github/jhrcook/KrasAlleleCna?branch=master"><img src="https://codecov.io/gh/jhrcook/KrasAlleleCna/branch/master/graph/badge.svg" alt="Coverage status"></a></li>
</ul>
</div>
</div>

</div>


      <footer><div class="copyright">
  <p>Developed by Joshua H Cook.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
