<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>B.02 - Parsing mpileup Results • KrasAlleleCna</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="B.02 - Parsing mpileup Results">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">KrasAlleleCna</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a01-download-getgdcdata.html">A.01 - Downloading Data</a>
    </li>
    <li>
      <a href="../articles/b01-dataprep-processannovar.html">B.01 - Parsing ANNOVAR Results</a>
    </li>
    <li>
      <a href="../articles/b02-dataprep-processmpileup.html">B.02 - Parsing mpileup Results</a>
    </li>
    <li>
      <a href="../articles/c01-calccopynum.html">C.01 - Calculate Allele-specific Copy Number</a>
    </li>
    <li>
      <a href="../articles/d01-plotresults.html">D.01 - Plot Allele-specific Copy Number Results</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jhrcook/KrasAlleleCna">
    <span class="fab fa fab fa-github fa-lg"></span>
     
    GitHub
  </a>
</li>
<li>
  <a href="https://www.haigislab.org">
    <span class="fas fa fas fa-flask fa-lg"></span>
     
    Haigis Lab
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>B.02 - Parsing mpileup Results</h1>
                        <h4 class="author">Joshua H Cook</h4>
            
            <h4 class="date">2019-04-27</h4>
      
      
      <div class="hidden name"><code>b02-dataprep-processmpileup.Rmd</code></div>

    </div>

    
    
<div id="set-up" class="section level1">
<h1 class="hasAnchor">
<a href="#set-up" class="anchor"></a>Set up</h1>
<p>Load libraries</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(tibble)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(stringr)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(KrasAlleleCna)</a></code></pre></div>
<p>Establish path variables:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">pkg_dir &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">"KrasAlleleCna"</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">extdata_dir &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="dt">package =</span> <span class="st">"KrasAlleleCna"</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">data_dir &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"data"</span>, <span class="dt">package =</span> <span class="st">"KrasAlleleCna"</span>)</a></code></pre></div>
<div id="tidy-vcf-files" class="section level2">
<h2 class="hasAnchor">
<a href="#tidy-vcf-files" class="anchor"></a>Tidy VCF files</h2>
<p>‘mpileup’ returned a separate file for each tissue sample (stored in “data-raw/gdc/mpileup_output”). In “data-raw/handle_mpileup.R”, each file was read in as a tibble, and a list of tibbles was saved to “inst/extdata/”.</p>
<p>This list of tibbles was read in and was combined into a single tibble, adding a new column with the file name. The <code>filename</code> column was parsed to create <code>file_id</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># a list of all mpileup files</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">mpileup_data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/readRDS">readRDS</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/file.path">file.path</a></span>(extdata_dir, <span class="st">"mpileup_output_list.rds"</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span>(<span class="dt">.id =</span> <span class="st">"filename"</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">file_id =</span> <span class="kw"><a href="https://stringr.tidyverse.org/reference/str_remove.html">str_remove_all</a></span>(filename, <span class="st">"_mpileup.vcf$"</span>))</a></code></pre></div>
<p>I only kept the VCF files for which the ANNOVAR data was successfully collected and contained a <em>KRAS</em> mutation.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># list of file IDs (after removing the "mpileup.vcf" suffix)</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2">mpileup_ids &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>(mpileup_data<span class="op">$</span>file_id))</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co"># number of shared IDs</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sum">sum</a></span>(mpileup_ids <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>(annovar_tib<span class="op">$</span>file_id))</a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt; [1] 182</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co"># number of VCF files removed in this step</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sum">sum</a></span>(<span class="op">!</span>(mpileup_ids <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>(annovar_tib<span class="op">$</span>file_id)))</a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt; [1] 222</span></a></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">mpileup_data &lt;-<span class="st"> </span>mpileup_data <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(file_id <span class="op">%in%</span><span class="st"> </span>annovar_tib<span class="op">$</span>file_id)</a></code></pre></div>
<p>The downstream steps only required the total number of reads and the number of mutant reads. These values were isolated into columns <code>total_num_reads</code> and <code>num_mut_reads</code>, respectively.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">mpileup_data &lt;-<span class="st"> </span>mpileup_data <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">total_num_reads =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(num_high_qual_bases),</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">               <span class="dt">num_mut_reads =</span> <span class="kw"><a href="https://stringr.tidyverse.org/reference/str_split.html">str_split_fixed</a></span>(allelic_depths,</a>
<a class="sourceLine" id="cb6-4" data-line-number="4">                                               <span class="st">","</span>, <span class="dv">2</span>)[, <span class="dv">2</span>]) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(file_id, start, total_num_reads, num_mut_reads) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="op">!</span>(<span class="kw"><a href="https://stringr.tidyverse.org/reference/str_detect.html">str_detect</a></span>(num_mut_reads, <span class="st">","</span>) <span class="op">|</span><span class="st"> </span>num_mut_reads <span class="op">==</span><span class="st"> ""</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">num_mut_reads =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(num_mut_reads),</a>
<a class="sourceLine" id="cb6-8" data-line-number="8">           <span class="dt">total_num_reads =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(total_num_reads))</a>
<a class="sourceLine" id="cb6-9" data-line-number="9">mpileup_data</a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt; # A tibble: 52,161 x 4</span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt;    file_id                              start total_num_reads num_mut_reads</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt;    &lt;chr&gt;                                &lt;dbl&gt;           &lt;dbl&gt;         &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt;  1 00b7ee04-b97b-49ee-b27b-13f6fb981…  2.52e7               1             1</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt;  2 00b7ee04-b97b-49ee-b27b-13f6fb981…  2.52e7               1             1</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt;  3 00b7ee04-b97b-49ee-b27b-13f6fb981…  2.52e7               2             1</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt;  4 00b7ee04-b97b-49ee-b27b-13f6fb981…  2.52e7               1             1</span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="co">#&gt;  5 00b7ee04-b97b-49ee-b27b-13f6fb981…  2.52e7               4             1</span></a>
<a class="sourceLine" id="cb6-18" data-line-number="18"><span class="co">#&gt;  6 00b7ee04-b97b-49ee-b27b-13f6fb981…  2.52e7               4             1</span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="co">#&gt;  7 00b7ee04-b97b-49ee-b27b-13f6fb981…  2.52e7               6             1</span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20"><span class="co">#&gt;  8 00b7ee04-b97b-49ee-b27b-13f6fb981…  2.52e7               3             1</span></a>
<a class="sourceLine" id="cb6-21" data-line-number="21"><span class="co">#&gt;  9 00b7ee04-b97b-49ee-b27b-13f6fb981…  2.52e7               1             1</span></a>
<a class="sourceLine" id="cb6-22" data-line-number="22"><span class="co">#&gt; 10 00b7ee04-b97b-49ee-b27b-13f6fb981…  2.52e7               1             1</span></a>
<a class="sourceLine" id="cb6-23" data-line-number="23"><span class="co">#&gt; # … with 52,151 more rows</span></a></code></pre></div>
<p>The last step was to merge the read-depth with the ANNOVAR mutation calls.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">allele_depth_tib &lt;-<span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/join.html">left_join</a></span>(annovar_tib, mpileup_data,</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">                              <span class="dt">by =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"file_id"</span> =<span class="st"> "file_id"</span>,</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">                                     <span class="st">"Start"</span> =<span class="st"> "start"</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/join.html">left_join</a></span>(tidy_tcga_sample_sheet, <span class="dt">by =</span> <span class="st">"file_id"</span>)</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">usethis<span class="op">::</span><span class="kw"><a href="https://usethis.r-lib.org/reference/use_data.html">use_data</a></span>(allele_depth_tib, <span class="dt">overwrite =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt; ✔ Setting active project to '/n/data2/bidmc/medicine/haigis/Cook/testAnalysisToPkg/KrasAlleleCna'</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt; ✔ Saving 'allele_depth_tib' to 'data/allele_depth_tib.rda'</span></a></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#set-up">Set up</a><ul class="nav nav-pills nav-stacked">
<li><a href="#tidy-vcf-files">Tidy VCF files</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Joshua H Cook.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
