<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>C.01 - Calculate Allele-specfic Copy Number • KrasAlleleCna</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="C.01 - Calculate Allele-specfic Copy Number">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">KrasAlleleCna</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a01-download-getgdcdata.html">A.01 - Downloading Data</a>
    </li>
    <li>
      <a href="../articles/b01-dataprep-processannovar.html">B.01 - Parsing ANNOVAR Results</a>
    </li>
    <li>
      <a href="../articles/b02-dataprep-processmpileup.html">B.02 - Parsing mpileup Results</a>
    </li>
    <li>
      <a href="../articles/c01-calccopynum.html">C.01 - Calculate Allele-specfic Copy Number</a>
    </li>
    <li>
      <a href="../articles/d01-plotresults.html">D.01 - Plot Allele-specific Copy Number Results</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>C.01 - Calculate Allele-specfic Copy Number</h1>
            
      
      
      <div class="hidden name"><code>c01-calccopynum.Rmd</code></div>

    </div>

    
    
<div id="set-up" class="section level1">
<h1 class="hasAnchor">
<a href="#set-up" class="anchor"></a>Set up</h1>
<p>Load libraries</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(readr)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(tibble)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(stringr)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(magrittr)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(KrasAlleleCna)</a></code></pre></div>
<p>Establish path variables:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">pkg_dir &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">"KrasAlleleCna"</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">extdata_dir &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="dt">package =</span> <span class="st">"KrasAlleleCna"</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">data_dir &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"data"</span>, <span class="dt">package =</span> <span class="st">"KrasAlleleCna"</span>)</a></code></pre></div>
<p>Below are the <em>KRAS</em> gene locations (it is on the negative strand).</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">KRAS_LOWER &lt;-<span class="st"> </span><span class="dv">25204789</span>  <span class="co"># end of gene</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">KRAS_UPPER &lt;-<span class="st"> </span><span class="dv">25251003</span>  <span class="co"># start of gene</span></a></code></pre></div>
</div>
<div id="load-and-prepare-data" class="section level1">
<h1 class="hasAnchor">
<a href="#load-and-prepare-data" class="anchor"></a>Load and prepare data</h1>
<p>The tumor purity was calculated by TCGA and downloaded from <a href="https://gdc.cancer.gov/about-data/publications/pancanatlas" class="uri">https://gdc.cancer.gov/about-data/publications/pancanatlas</a> where it was labeled as “ABSOLUTE purity/ploidy file - TCGA_mastercalls.abs_tables_JSedit.fixed.txt”. The only data required for downstream analysis is the ID (below, extracted from the <code>sample</code> column) and the purity.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">purity_tib &lt;-<span class="st"> </span><span class="kw"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_tsv</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/file.path">file.path</a></span>(extdata_dir, <span class="st">"tcga_purity_ploidy.tsv"</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">common_id =</span> <span class="kw"><a href="https://stringr.tidyverse.org/reference/str_sub.html">str_sub</a></span>(sample, <span class="dv">1</span>, <span class="dv">16</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(common_id, purity)</a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#&gt; Parsed with column specification:</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt; cols(</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co">#&gt;   array = col_character(),</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt;   sample = col_character(),</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt;   `call status` = col_character(),</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt;   purity = col_double(),</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt;   ploidy = col_double(),</span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt;   `Genome doublings` = col_double(),</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt;   `Coverage for 80% power` = col_double(),</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="co">#&gt;   `Cancer DNA fraction` = col_double(),</span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#&gt;   `Subclonal genome fraction` = col_double(),</span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co">#&gt;   solution = col_character()</span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co">#&gt; )</span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17">purity_tib</a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="co">#&gt; # A tibble: 10,786 x 2</span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="co">#&gt;    common_id        purity</span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"><span class="co">#&gt;    &lt;chr&gt;             &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="co">#&gt;  1 TCGA-OR-A5J1-01A   0.9 </span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="co">#&gt;  2 TCGA-OR-A5J2-01A   0.89</span></a>
<a class="sourceLine" id="cb4-23" data-line-number="23"><span class="co">#&gt;  3 TCGA-OR-A5J3-01A   0.93</span></a>
<a class="sourceLine" id="cb4-24" data-line-number="24"><span class="co">#&gt;  4 TCGA-OR-A5J4-01A   0.87</span></a>
<a class="sourceLine" id="cb4-25" data-line-number="25"><span class="co">#&gt;  5 TCGA-OR-A5J5-01A   0.93</span></a>
<a class="sourceLine" id="cb4-26" data-line-number="26"><span class="co">#&gt;  6 TCGA-OR-A5J6-01A   0.69</span></a>
<a class="sourceLine" id="cb4-27" data-line-number="27"><span class="co">#&gt;  7 TCGA-OR-A5J7-01A   0.84</span></a>
<a class="sourceLine" id="cb4-28" data-line-number="28"><span class="co">#&gt;  8 TCGA-OR-A5J8-01A   0.76</span></a>
<a class="sourceLine" id="cb4-29" data-line-number="29"><span class="co">#&gt;  9 TCGA-OR-A5J9-01A   0.84</span></a>
<a class="sourceLine" id="cb4-30" data-line-number="30"><span class="co">#&gt; 10 TCGA-OR-A5JA-01A   0.75</span></a>
<a class="sourceLine" id="cb4-31" data-line-number="31"><span class="co">#&gt; # … with 10,776 more rows</span></a></code></pre></div>
<p>Tumor CNV data were downloaded from <a href="https://gdc.cancer.gov/about-data/publications/pancanatlas" class="uri">https://gdc.cancer.gov/about-data/publications/pancanatlas</a> and labeled “Copy Number - broad.mit.edu_PANCAN_Genome_Wide_SNP_6_whitelisted.seg” (Due to file size restrictions, “tcga_copynumber.tsv” is available in a compressed for in “inst/extdata/”, and was loaded into “data” for use in the package using <code><a href="https://usethis.r-lib.org/reference/use_data.html">usethis::use_data()</a></code>.)</p>
<p><em>KRAS</em> is on chromosome 12 in humans, so only data for that chromomsome was retained. The <code>common_id</code> was extracted as above and the <code>start</code> and <code>stop</code> columns for the CNV calls were renamed to be more specific to remove ambiguity after merging data sets, later. Finally, the copy number was calculated from the <code>Segment_Mean</code>; I found out how to do this <a href="https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/CNV_Pipeline/">here</a>: “The GDC further transforms these copy number values into segment mean values, which are equal to <strong>log2(copy-number/ 2)</strong>. Diploid regions will have a segment mean of zero, amplified regions will have positive values, and deletions will have negative values.”</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">cnv_tib <span class="op">%&lt;&gt;%</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(Chromosome <span class="op">==</span><span class="st"> </span><span class="dv">12</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">common_id =</span> <span class="kw"><a href="https://stringr.tidyverse.org/reference/str_sub.html">str_sub</a></span>(Sample, <span class="dv">1</span>, <span class="dv">16</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">rename</a></span>(<span class="dt">CNV_start =</span> Start,</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">                  <span class="dt">CNV_end =</span> End) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">copy_number =</span> <span class="dv">2</span><span class="op">^</span>(Segment_Mean <span class="op">+</span><span class="st"> </span><span class="dv">1</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(common_id, CNV_start, CNV_end, copy_number)</a>
<a class="sourceLine" id="cb5-8" data-line-number="8">cnv_tib</a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt; # A tibble: 125,292 x 4</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co">#&gt;    common_id        CNV_start   CNV_end copy_number</span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">#&gt;    &lt;chr&gt;                &lt;dbl&gt;     &lt;dbl&gt;       &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="co">#&gt;  1 TCGA-OR-A5J1-10A    889902 133161346        2.00</span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13"><span class="co">#&gt;  2 TCGA-OR-A5J1-01A    889902   6397137        2.03</span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="co">#&gt;  3 TCGA-OR-A5J1-01A   6405578   6604354        2.88</span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="co">#&gt;  4 TCGA-OR-A5J1-01A   6605105  69115694        2.01</span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16"><span class="co">#&gt;  5 TCGA-OR-A5J1-01A  69116140  70403918        2.88</span></a>
<a class="sourceLine" id="cb5-17" data-line-number="17"><span class="co">#&gt;  6 TCGA-OR-A5J1-01A  70413723 109705218        2.01</span></a>
<a class="sourceLine" id="cb5-18" data-line-number="18"><span class="co">#&gt;  7 TCGA-OR-A5J1-01A 109707986 110609714        2.94</span></a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="co">#&gt;  8 TCGA-OR-A5J1-01A 110619349 113176543        2.02</span></a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="co">#&gt;  9 TCGA-OR-A5J1-01A 113178406 113622306        2.96</span></a>
<a class="sourceLine" id="cb5-21" data-line-number="21"><span class="co">#&gt; 10 TCGA-OR-A5J1-01A 113626419 113881057        3.84</span></a>
<a class="sourceLine" id="cb5-22" data-line-number="22"><span class="co">#&gt; # … with 125,282 more rows</span></a></code></pre></div>
<p>Both the purity and copy number data was joined to the ANNOVAR and mpileup results.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">allele_data &lt;-<span class="st"> </span>allele_depth_tib <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">common_id =</span> sample_id) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(common_id, Chr, Start, End, aa_mod, total_num_reads, num_mut_reads,</a>
<a class="sourceLine" id="cb6-4" data-line-number="4">           Ref, Alt, Func.refGene, Gene.refGene, ExonicFunc.refGene, file_name,</a>
<a class="sourceLine" id="cb6-5" data-line-number="5">           file_id, project_id, case_id, sample_id, sample_type) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/join.html">left_join</a></span>(cnv_tib, <span class="dt">by =</span> <span class="st">"common_id"</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/join.html">left_join</a></span>(purity_tib, <span class="dt">by =</span> <span class="st">"common_id"</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>()</a></code></pre></div>
</div>
<div id="gather-cnv-values" class="section level1">
<h1 class="hasAnchor">
<a href="#gather-cnv-values" class="anchor"></a>Gather CNV values</h1>
<p>Some samples had multiple measures of copy number along <em>KRAS</em>. The following section selects a single value for each sample. It isn’t pretty, but gets the job done.</p>
<p>An empty tibble was created with the correct column names for which the data will be add to row-wise</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">allele_data_filt &lt;-<span class="st"> </span>allele_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice</a></span>(<span class="dv">0</span>)</a></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="cf">for</span> (fi <span class="cf">in</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(allele_data<span class="op">$</span>file_id)) {</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">    <span class="co"># temp tib with this loops data</span></a>
<a class="sourceLine" id="cb8-3" data-line-number="3">    tt &lt;-<span class="st"> </span>allele_data <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="st">        </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(file_id <span class="op">==</span><span class="st"> </span>fi)</a>
<a class="sourceLine" id="cb8-5" data-line-number="5">    <span class="co"># if WT:</span></a>
<a class="sourceLine" id="cb8-6" data-line-number="6">    <span class="cf">if</span> (tt<span class="op">$</span>aa_mod[[<span class="dv">1</span>]] <span class="op">==</span><span class="st"> "WT"</span>) {</a>
<a class="sourceLine" id="cb8-7" data-line-number="7">        <span class="co"># bool for if CNV probe straddles or cuts KRAS</span></a>
<a class="sourceLine" id="cb8-8" data-line-number="8">        tt <span class="op">%&lt;&gt;%</span></a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="st">            </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">straddle_kras =</span> (CNV_start <span class="op">&lt;</span><span class="st"> </span>KRAS_LOWER <span class="op">&amp;</span></a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="st">                                    </span>KRAS_UPPER <span class="op">&lt;</span><span class="st"> </span>CNV_end),</a>
<a class="sourceLine" id="cb8-11" data-line-number="11">                   <span class="dt">cuts_kras =</span> ((CNV_start <span class="op">&lt;</span><span class="st"> </span>KRAS_LOWER <span class="op">&amp;</span></a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="st">                                </span>CNV_end <span class="op">&lt;</span><span class="st"> </span>KRAS_UPPER) <span class="op">|</span></a>
<a class="sourceLine" id="cb8-13" data-line-number="13"><span class="st">                               </span>(KRAS_LOWER <span class="op">&lt;</span><span class="st"> </span>CNV_start <span class="op">&amp;</span></a>
<a class="sourceLine" id="cb8-14" data-line-number="14"><span class="st">                                </span>KRAS_UPPER <span class="op">&lt;</span><span class="st"> </span>CNV_end)),</a>
<a class="sourceLine" id="cb8-15" data-line-number="15">                   <span class="dt">dist_to_kras =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Extremes">min</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">abs</a></span>(CNV_start <span class="op">-</span><span class="st"> </span>KRAS_LOWER),</a>
<a class="sourceLine" id="cb8-16" data-line-number="16">                                      <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">abs</a></span>(CNV_start <span class="op">-</span><span class="st"> </span>KRAS_UPPER),</a>
<a class="sourceLine" id="cb8-17" data-line-number="17">                                      <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">abs</a></span>(CNV_end <span class="op">-</span><span class="st"> </span>KRAS_LOWER),</a>
<a class="sourceLine" id="cb8-18" data-line-number="18">                                      <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">abs</a></span>(CNV_end <span class="op">-</span><span class="st"> </span>KRAS_UPPER)))</a>
<a class="sourceLine" id="cb8-19" data-line-number="19">        <span class="co"># prefer straddle</span></a>
<a class="sourceLine" id="cb8-20" data-line-number="20">        <span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/any">any</a></span>(tt<span class="op">$</span>straddle_kras, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)) {</a>
<a class="sourceLine" id="cb8-21" data-line-number="21">            tt <span class="op">%&lt;&gt;%</span></a>
<a class="sourceLine" id="cb8-22" data-line-number="22"><span class="st">                </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(straddle_kras)</a>
<a class="sourceLine" id="cb8-23" data-line-number="23">        <span class="co"># else use probes that cut KRAS</span></a>
<a class="sourceLine" id="cb8-24" data-line-number="24">        } <span class="cf">else</span> <span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/any">any</a></span>(tt<span class="op">$</span>cuts_kras, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)) {</a>
<a class="sourceLine" id="cb8-25" data-line-number="25">            tt <span class="op">%&lt;&gt;%</span></a>
<a class="sourceLine" id="cb8-26" data-line-number="26"><span class="st">                </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(cuts_kras)</a>
<a class="sourceLine" id="cb8-27" data-line-number="27">        } <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb8-28" data-line-number="28">            <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">message</a></span>(<span class="st">"insufficient filters"</span>)</a>
<a class="sourceLine" id="cb8-29" data-line-number="29">        }</a>
<a class="sourceLine" id="cb8-30" data-line-number="30">        <span class="co"># standardize tt to be rbind</span></a>
<a class="sourceLine" id="cb8-31" data-line-number="31">        tt  <span class="op">%&lt;&gt;%</span></a>
<a class="sourceLine" id="cb8-32" data-line-number="32"><span class="st">            </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">copy_number =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/mean">mean</a></span>(copy_number)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-33" data-line-number="33"><span class="st">            </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice</a></span>(<span class="dv">1</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-34" data-line-number="34"><span class="st">            </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="op">-</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"straddle_kras"</span>, <span class="st">"cuts_kras"</span>, <span class="st">"dist_to_kras"</span>))</a>
<a class="sourceLine" id="cb8-35" data-line-number="35">    } <span class="cf">else</span> {  <span class="co"># if mutant:</span></a>
<a class="sourceLine" id="cb8-36" data-line-number="36">        <span class="co"># keep CNV probe that contains the mutation</span></a>
<a class="sourceLine" id="cb8-37" data-line-number="37">        tt <span class="op">%&lt;&gt;%</span></a>
<a class="sourceLine" id="cb8-38" data-line-number="38"><span class="st">            </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(CNV_start <span class="op">&lt;</span><span class="st"> </span>Start <span class="op">&amp;</span><span class="st"> </span>Start <span class="op">&lt;</span><span class="st"> </span>CNV_end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-39" data-line-number="39"><span class="st">            </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">copy_number =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/mean">mean</a></span>(copy_number)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-40" data-line-number="40"><span class="st">            </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice</a></span>(<span class="dv">1</span>)</a>
<a class="sourceLine" id="cb8-41" data-line-number="41">        <span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">nrow</a></span>(tt) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>) {</a>
<a class="sourceLine" id="cb8-42" data-line-number="42">            <span class="co"># if no probe contained it, just return the original data and warn</span></a>
<a class="sourceLine" id="cb8-43" data-line-number="43">            tt &lt;-<span class="st"> </span>allele_data <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-44" data-line-number="44"><span class="st">                </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(file_id <span class="op">==</span><span class="st"> </span>fi) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-45" data-line-number="45"><span class="st">                </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice</a></span>(<span class="dv">1</span>)</a>
<a class="sourceLine" id="cb8-46" data-line-number="46">            <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/warning">warning</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(<span class="st">"Mutant with no rows:"</span>, fi))</a>
<a class="sourceLine" id="cb8-47" data-line-number="47">        }</a>
<a class="sourceLine" id="cb8-48" data-line-number="48">    }</a>
<a class="sourceLine" id="cb8-49" data-line-number="49">    <span class="co"># compile tt into the final tibble</span></a>
<a class="sourceLine" id="cb8-50" data-line-number="50">    allele_data_filt &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span>(allele_data_filt, tt)</a>
<a class="sourceLine" id="cb8-51" data-line-number="51">}</a>
<a class="sourceLine" id="cb8-52" data-line-number="52"><span class="co">#&gt; Warning: Mutant with no rows: 1da80f23-c59f-4ace-ba93-a884e49a0ca5</span></a>
<a class="sourceLine" id="cb8-53" data-line-number="53"><span class="co">#&gt; Warning: Mutant with no rows: 6fb1db2f-7e6f-4aaa-bdaf-a59fb9e89275</span></a>
<a class="sourceLine" id="cb8-54" data-line-number="54"><span class="co">#&gt; Warning: Mutant with no rows: 889ecef7-f701-4b3c-898e-f43116fa9f48</span></a>
<a class="sourceLine" id="cb8-55" data-line-number="55"><span class="co">#&gt; Warning: Mutant with no rows: 92147064-5421-421e-9ac1-c076e0d8782a</span></a>
<a class="sourceLine" id="cb8-56" data-line-number="56"><span class="co">#&gt; Warning: Mutant with no rows: 971c7d88-323c-460c-9143-c4e0c5e9aad8</span></a>
<a class="sourceLine" id="cb8-57" data-line-number="57"><span class="co">#&gt; Warning: Mutant with no rows: abec8d73-05fb-4d16-b462-728993dca9ba</span></a>
<a class="sourceLine" id="cb8-58" data-line-number="58"><span class="co">#&gt; Warning: Mutant with no rows: b995ca5b-cca3-4819-bb0b-cd087e2d7722</span></a></code></pre></div>
<p>Check if any mutants have missing data.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">allele_data_filt <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/NA">is.na</a></span>(purity) <span class="op">&amp;</span><span class="st"> </span>aa_mod <span class="op">!=</span><span class="st"> "WT"</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(aa_mod, project_id)</a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#&gt; # A tibble: 13 x 2</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="co">#&gt;    aa_mod project_id</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;     </span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="co">#&gt;  1 G12D   TCGA-COAD </span></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co">#&gt;  2 G12D   TCGA-COAD </span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">#&gt;  3 G12D   TCGA-COAD </span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="co">#&gt;  4 Q61K   TCGA-COAD </span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co">#&gt;  5 G13D   TCGA-COAD </span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="co">#&gt;  6 G12V   TCGA-COAD </span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="co">#&gt;  7 G12A   TCGA-COAD </span></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="co">#&gt;  8 G13D   TCGA-COAD </span></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="co">#&gt;  9 G12D   TCGA-COAD </span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="co">#&gt; 10 G12C   TCGA-COAD </span></a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="co">#&gt; 11 G12D   TCGA-COAD </span></a>
<a class="sourceLine" id="cb9-18" data-line-number="18"><span class="co">#&gt; 12 G12D   TCGA-COAD </span></a>
<a class="sourceLine" id="cb9-19" data-line-number="19"><span class="co">#&gt; 13 G12D   TCGA-COAD</span></a></code></pre></div>
</div>
<div id="calculate-allele-specific-copy-number" class="section level1">
<h1 class="hasAnchor">
<a href="#calculate-allele-specific-copy-number" class="anchor"></a>Calculate allele specific copy number</h1>
<p>Finally, the allele specific copy number was calculated. The equation was taken from Eq. (2) in Stephens et al. 2012, Nature Letter:</p>
<pre><code>n_mut = (r/R) * (1/p) * (p * n_locus + 2 * (1 - p))</code></pre>
<p>where n_mut = copy number of the mutant allele r = number of reads with a mutation R = number of reads total from the locus p = fraction of tumor cells (i.e. purity) n_locus = copy number of the locus</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">allele_data_filt <span class="op">%&lt;&gt;%</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">cn_mut =</span> (num_mut_reads <span class="op">/</span><span class="st"> </span>(total_num_reads <span class="op">*</span><span class="st"> </span>purity)) <span class="op">*</span></a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="st">           </span>(purity <span class="op">*</span><span class="st"> </span>copy_number <span class="op">+</span><span class="st"> </span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>purity))),</a>
<a class="sourceLine" id="cb11-4" data-line-number="4">           <span class="dt">copy_number_adj =</span> (<span class="dv">1</span> <span class="op">/</span><span class="st"> </span>purity) <span class="op">*</span><span class="st"> </span>(copy_number <span class="op">-</span><span class="st"> </span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>purity))),</a>
<a class="sourceLine" id="cb11-5" data-line-number="5">           <span class="dt">cn_wt =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/ifelse">ifelse</a></span>(aa_mod <span class="op">==</span><span class="st"> "WT"</span>,</a>
<a class="sourceLine" id="cb11-6" data-line-number="6">                          copy_number_adj,</a>
<a class="sourceLine" id="cb11-7" data-line-number="7">                          copy_number_adj <span class="op">-</span><span class="st"> </span>cn_mut))</a></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">usethis<span class="op">::</span><span class="kw"><a href="https://usethis.r-lib.org/reference/use_data.html">use_data</a></span>(allele_data_filt, <span class="dt">overwrite =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="co">#&gt; ✔ Setting active project to '/n/data2/bidmc/medicine/haigis/Cook/testAnalysisToPkg/KrasAlleleCna'</span></a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="co">#&gt; ✔ Saving 'allele_data_filt' to 'data/allele_data_filt.rda'</span></a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#set-up">Set up</a></li>
      <li><a href="#load-and-prepare-data">Load and prepare data</a></li>
      <li><a href="#gather-cnv-values">Gather CNV values</a></li>
      <li><a href="#calculate-allele-specific-copy-number">Calculate allele specific copy number</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Joshua H Cook.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
